ENIGMAsystem/SHELL/Graphics_Systems/Direct3D9/DX9screen.cpp

```diff
@@ -151,9 +152,8 @@ static inline int draw_tiles()
   {
     if (dit->second.tiles.size())
     {
-      for (unsigned int t = 0; t<drawing_depths[dit->second.tiles[0].depth].tilevector.size(); ++t){
-        enigma_user::texture_set(drawing_depths[dit->second.tiles[0].depth].tilevector[t][0]);
-        //d3d_model_part_draw(drawing_depths[dit->second.tiles[0].depth].tilelist, drawing_depths[dit->second.tiles[0].depth].tilevector[t][1], drawing_depths[dit->second.tiles[0].depth].tilevector[t][2]);
+      for (auto &t : tile_layer_metadata[dit->second.tiles[0].depth]){
```

Comment:
@JoshDreamland will cry bout you using auto prolly

Reply:
I just copied Harri's code there from OpenGL3 and changed it because he used d3d_model functions whereas polygonz original code used glCallLists (so his was easier to "generify" basically). So, yeah, I can change it, it's whatever.

Reply:
No, auto is fine. In this case, it's pretty clear what `t` is.